### PR TITLE
Introduce lambda function and package scripts

### PR DESCRIPTION
**Pull request by monstrobishi**
#### What kind of PR is this?:

/kind

#### What this PR does / why we need it:

Introduce lambda function and add package scripts

#### Which issue(s) does this PR fixes?:

Fixes #

#### Additional comments?:

The lambda function here is mostly placeholder, the main purpose is the monorepo packaging

### CODE REVIEW

**Review by jingyi2811 — APPROVED** [review]

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,8 @@ npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
 lerna-debug.log*
+
+# Build
+build
+bundle.zip
+packages/**/package-lock.json
```
Comment (fuxingloh) [inline .gitignore:52]: I don't think `package-lock` should be ignored, it makes it easy to spot whether the developer is using the mono repo correctly.

File: package.json [hunk 2]
```diff
@@ -18,7 +18,8 @@
     "standard": "ts-standard --fix",
     "test": "jest",
     "test:ci": "jest --ci --coverage --forceExit --maxWorkers=4",
-    "all": "npm run build && npm run standard && npm run test"
+    "all": "npm run build && npm run standard && npm run test",
+    "package": "npm run pack -w ./packages/salmon-lambda-functions"
```
Comment (fuxingloh) [inline package.json:22]: we still need this?

File: packages/salmon-lambda-functions/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { PriceManager, PriceSourceConfig, TiingoPriceProvider } from '@defichain/salmon-price-functions'
```
Comment (fuxingloh) [inline packages/salmon-lambda-functions/src/index.ts:1]: future refactor opportunity for better maintainability, to separate out all providers into a package `salmon-price-providers`

Maybe can create an issue for this

Comment (fuxingloh) [reply packages/salmon-lambda-functions/src/index.ts:1]: On another thought, maybe we can have separate it into `salmon-lambda-tiingo`? making each package a lambda function that is deployable. This keeps the concerns extremely separated, making the core acts more like a core that is provider agnostic.

**Review by fuxingloh — COMMENTED** [review]
thinking long term here, left some thoughts about it

File: package.json (hunk 2 above)
Comment (monstrobishi) [reply package.json:22]: This just handy for running package from root dir, which in turn calls ncc on that module

File: packages/salmon-lambda-functions/src/index.ts (hunk 3 above)
Comment (monstrobishi) [reply packages/salmon-lambda-functions/src/index.ts:1]: Sure I like the second option, will open a ticket for this

**Review by fuxingloh — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (fuxingloh) [reply package.json:22]: oh, I didn't see `-w`

**Review by canonbrother — APPROVED** [review]

File: packages/salmon-lambda-functions/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export async function handler (event?: any): Promise<any> {
+  const tiingoConfig: PriceSourceConfig = {
+    symbols: ['TSLA']
+  }
+
+  const priceManager = new PriceManager(tiingoConfig, new TiingoPriceProvider(process.env.API_TOKEN ?? ''))
+  const prices = await priceManager.fetchAssetPrices()
+
+  console.log(JSON.stringify({ prices, event }))
```
Comment (jingyi2811) [inline packages/salmon-lambda-functions/src/index.ts:11]: You probably need to remove this line

Comment (monstrobishi) [reply packages/salmon-lambda-functions/src/index.ts:11]: This is intentional for logging inside the lambda function, I can change to a better logging function in a future PR

**Review by jingyi2811 — APPROVED** [review]

**Pull request merged**